BlueDBOnDisk/src/main/java/org/bluedb/disk/collection/index/ReadWriteBlueIndexOnDisk.java

```diff
@@ -60,43 +57,60 @@
 		}
 	}
 
-	public static <K extends ValueKey, T extends Serializable> BlueIndexOnDisk<K, T> fromExisting(ReadOnlyBlueCollectionOnDisk<T> collection, Path indexPath) throws BlueDbException {
-		FileManager fileManager = collection.getFileManager();
+	public static <K extends ValueKey, T extends Serializable> ReadWriteBlueIndexOnDisk<K, T> fromExisting(ReadWriteBlueCollectionOnDisk<T> collection, Path indexPath) throws BlueDbException {
+		ReadWriteFileManager fileManager = collection.getFileManager();
 		Path keyExtractorPath = Paths.get(indexPath.toString(), FILE_KEY_EXTRACTOR);
 		@SuppressWarnings("unchecked")
 		KeyExtractor<K, T> keyExtractor = (KeyExtractor<K, T>) fileManager.loadObject(keyExtractorPath);
-		return new BlueIndexOnDisk<K, T>(collection, indexPath, keyExtractor);
-	}
-
-	public Class<I> getType() {
-		return keyExtractor.getType();
+		return new ReadWriteBlueIndexOnDisk<K, T>(collection, indexPath, keyExtractor);
 	}
 
-	private BlueIndexOnDisk(ReadOnlyBlueCollectionOnDisk<T> collection, Path indexPath, KeyExtractor<I, T> keyExtractor) throws BlueDbException {
-		this.collection = collection;
-		this.keyExtractor = keyExtractor;
-		this.fileManager = collection.getFileManager();
+	private ReadWriteBlueIndexOnDisk(ReadWriteBlueCollectionOnDisk<T> collection, Path indexPath, KeyExtractor<I, T> keyExtractor) throws BlueDbException {
+		super(collection, indexPath, keyExtractor);
 		this.indexName = indexPath.toFile().getName();
+		this.fileManager = collection.getFileManager();
 		SegmentSizeSetting sizeSetting = determineSegmentSize(keyExtractor.getType());
-		segmentManager = new SegmentManager<BlueKey>(indexPath, fileManager, this, sizeSetting.getConfig());
+		segmentManager = new ReadWriteSegmentManager<BlueKey>(indexPath, fileManager, this, sizeSetting.getConfig());
+		rollupScheduler = ((ReadWriteBlueCollectionOnDisk<T>) collection).getRollupScheduler();
 	}
 
-	protected static SegmentSizeSetting determineSegmentSize(Class<? extends BlueKey> keyType) throws BlueDbException {
-		/*
-		 * TODO: This will change to use the default index setting once we support different index segment sizes. We
-		 * may require information from the meta data in order to make this determiniation. We may also need to
-		 * know if the index is new and if the user has requeste a particular size for this index. See BlueCollectionOnDisk#determineSegmentSize
-		 * for inspiration.
-		 */
-		return SegmentSizeSetting.getOriginalDefaultSettingsFor(keyType);
+	public ReadWriteSegmentManager<BlueKey> getSegmentManager() {
+		return segmentManager;
+	}
+
+	@Override
+	public void reportReads(List<RollupTarget> rollupTargets) {
+		if (rollupScheduler != null) {

```

Comment:
This null check is unnecessary. As far as I can tell, the rollup scheduler should always be set. Same goes for the null check on line 92

Reply:
removing these in the next commit